R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -2165,6 +2165,14 @@ test_that("SQL error message is returned from JVM", {
   expect_equal(grepl("blah", retError), TRUE)
 })
 
+test_that("Default warehouse dir should be set to tempdir", {
+  # nothing should be written outside tempdir() without explicit user permission
+  inital_working_directory_files <- list.files()
```

Comment:
Yeah. I think @felixcheung point is right - The dir should be created early on. Also I think in the tests we sometimes configure the `hive.metastore.dir` in our tests and so I dont see it come up when I run the test cases.  I'll try to see if we can design a different test case.